adafruit_azureiot/iot_mqtt.py

```diff
@@ -0,0 +1,453 @@
+"""MQTT client for Azure IoT
+"""
+
+import gc
+import json
+import time
+import adafruit_esp32spi.adafruit_esp32spi_socket as socket
+from adafruit_esp32spi.adafruit_esp32spi_wifimanager import ESPSPI_WiFiManager
```

Comment:
Instead of importing esp32spi and the esp32spi socket, could we the `socket` object in instead?

This will allow us to use multiple types of `socket` objects with this library like wifi, ethernet, and cellular.

We do this here in the latest `MiniMQTT`: https://github.com/adafruit/Adafruit_CircuitPython_MiniMQTT/blob/master/examples/minimqtt_simpletest_eth.py#L64

Reply:
Yup - I was following the pattern of the existing code, relying on the WiFi manager. Happy to change it to a socket instead!

Reply:
That'd be great. Let me know if you have any Q's. I'll test with ESP32 and Ethernet `socket`s.

Reply:
Updated to use the socket and interface